demo/ibis-demo.es.qmd

```diff
@@ -0,0 +1,258 @@
+    avg_rating=ratings.average_rating.cast("float"),
+    num_votes=ratings.num_votes.cast("int"),
+)
+```
+
+```{python}
+ratings = ratings.select(
+    ratings.tconst,
+    avg_rating=ratings.average_rating.cast("float"),
+    num_votes=ratings.num_votes.cast("int"),
+)
+```
+
+## `to_sql()`
+
+Y cuando quiere ver el SQL, tenemos el método `to_sql`:
+
+```{python}
+ibis.to_sql(ratings)
+```
+
+```{python}
+ibis.to_sql(ratings, dialect="sqlite")
+```
+
+```{python}
+ibis.options.interactive = False
+```
+
+## `.sql`
+
+Luego, tenemos el método `.sql()` y este método es para escribir SQL crudo:
+
+A veces, usted necesita escribir SQL.
+
+```{python}
+con.sql("""
+        SELECT
+        "t0"."tconst",
+        CAST("t0"."averageRating" AS VARCHAR) AS "avg_rating",
+        CAST("t0"."numVotes" AS VARCHAR) AS "num_votes"
+        FROM "imdb_title_ratings" AS "t0"
+    """)
+```
+
+```{python}
+con.sql("""
+        SELECT
+        "t0"."tconst",
+        CAST("t0"."averageRating" AS VARCHAR) AS "avg_rating",
+        CAST("t0"."numVotes" AS VARCHAR) AS "num_votes"
+        FROM "imdb_title_ratings" AS "t0"
+    """).execute()
+```
+
+```{python}
+ibis.options.interactive = True
+```
+
+```{python}
+con.sql("""
+        SELECT
+        "t0"."tconst",
+        CAST("t0"."averageRating" AS VARCHAR) AS "avg_rating",
+        CAST("t0"."numVotes" AS VARCHAR) AS "num_votes"
+        FROM "imdb_title_ratings" AS "t0"
+    """)
+```
+
+## Otras operaciones
+
+```{python}
+basics.columns
+```
+
+Para "GROUP BY" con `count()` tenemos el método `value_counts()`
+
+```{python}
+basics.title_type.value_counts()
+```
+
+Para nuestros analysis final, limipiaremos unas cosas como seleccionando solo titulos con el tipo "película" y desechando las películas adultos.
+
+```{python}
+basics = (
+    basics
+    .filter([basics.title_type == "movie", basics.is_adult == 0])
+    .select("tconst", "primary_title")
+)
+```
+
+```{python}
+basics
+```
+
+Ahora, unimos la tabla `basics` y la tabla `ratings` con la columna `tconst`, y ejecutamos.
+
+```{python}
+basics.join(ratings, "tconst").execute(limit=10)
+```
+
+Ordenamos por `avg_rating`
+
+```{python}
+basics.join(ratings, "tconst").order_by(_.avg_rating.desc())
+```
+
+Y filtramos titulos con votos menos que uno millón.
+
+```{python}
+basics.join(ratings, "tconst").order_by(_.avg_rating.desc()).filter(_.num_votes > 1e6)
+```
+
+## Unión de tablas (`join`)
+
+Finalmente, mostramos la expresión entera para computar las diez mejores películas.
+
+```{python}
+topfilms = basics.join(ratings, "tconst").order_by(_.avg_rating.desc()).filter(_.num_votes > 1e6)
+```
+
+```{python}
+ibis.options.interactive = False
+```
+
+```{python}
+topfilms
+```
+
+```{python}
+topfilms.execute(limit=10)
+```
+
+## ¡Ejecutamos la misma expresión en Postgres!
+
+```{python}
+pgcon = ibis.postgres.connect()
+pgcon.list_tables()
+```
+
+### Poner los datos adentro de Postgres
+
+```{python}
+!psql < create_imdb.sql
+!duckdb < load_imdb.sql
```

Comment:
Why do we need this step for duckdb?

Reply:
DuckDB is actually the one loading the data into postgres, because it's way easier to do it that way than to load it with postgres.

Reply:
Oh, sorry it's actually separate because duckdb doesn't respect quotes when creating tables in postgres. https://github.com/ibis-project/ibis-tutorial/pull/14/files#r1598860465